internal/services/containers/kubernetes_cluster_resource.go

```diff
@@ -3187,6 +3223,27 @@ func expandKubernetesClusterWorkloadAutoscalerProfile(input []interface{}, d *pl
 	return &workloadAutoscalerProfile
 }
 
+func expandKubernetesClusterNodeProvisioningProfile(input []interface{}) *managedclusters.ManagedClusterNodeProvisioningProfile {
+	if len(input) == 0 {
+		return nil
+	}
+	config := input[0].(map[string]interface{})
+	profile := &managedclusters.ManagedClusterNodeProvisioningProfile{}
+	if v := config["mode"].(string); v != "" {
+		mv := managedclusters.NodeProvisioningMode(v)
+		profile.Mode = &mv
+	}
+	if v := config["default_node_pools"].(string); v != "" {
+		dv := managedclusters.NodeProvisioningDefaultNodePools(v)
+		profile.DefaultNodePools = &dv
+	}
+	// If both fields are empty treat as nil to avoid sending empty object
```

Comment:
Instead of doing this, as well as to avoid the potential panic above, shall we instead add a constraint to say at least one of these attributes shall be specified, if the parent attribute is specified?